src/libs/snapshots.c

```diff
@@ -469,8 +497,15 @@ int mouse_moved(dt_lib_module_t *self,
     {
       d->vp_xpointer = xp;
       d->vp_ypointer = yp;
-      dt_control_queue_redraw_center();
     }
+
+    // Here to ensure the rotation symbol is light-up or light-off
+    // when moving close.
+    const gboolean display_rotation = _get_rotation_area(self, x, y, NULL, NULL, NULL);
+
+    if(d->dragging || display_rotation != d->rotsym_lightup)
```

Comment:
Why not _just_ calculate `d->rotsym_lightup` here (after temporarily saving the previous value) and then call redraw if the value has changed. In post_expose you can then just use the existing value of `d->rotsym_lightup` instead of doing the whole calculation again. Makes it less necessary/useful to spin off the calculation into a separate function.